### PR TITLE
Allow TLS command-line options to be set via environment variables

### LINKED ISSUE #3456: Error: unknown flag: --tls
Including the TLS command-line options on every call is tedious so I created a wrapper script that does:

    #!/bin/bash
    exec helm \
      --tls \
      --tls-verify \
      --tls-ca-cert=/path/to/ca.pem \
      --tls-cert=/path/to/cert.pem \
      --tls-key=/path/to/key.pem \
      "$@"
 
This works fine for commands like "helm ls" or "helm install" that communicate with Tiller but fails with *Error: unknown flag: --tls* with "helm", "helm search" or "helm repo list" for those that don't.  

Could this be changed to that these command-line options are allowed for all helm commands, but are ignored  when not communicating with Tiller?

Another alternative would be to add support for for a TILLER_TLS_* environment variables that are used like HELM_HOST or TILLER_NAMESPACE.  Then we could set the environment variables:

    export HELM_TLS_ENABLE=true
    export HELM_TLS_VERIFY=true
    export HELM_TLS_CA_CERT=/path/to/ca.pem
    export HELM_TLS_CERT=/path/to/cert.pem
    export HELM_TLS_KEY=/path/to/key.pem

and all helm commands would work without lots of command-line options.  Perhaps doing both would make sense.

Thoughts?

### PR DESCRIPTION
**Pull request by erwbgy**
Allow TLS command-line options to be set via environment variables and be used with all commands.

Add or update tests for these changes , remove duplicate --tls-ca-cert option in init and use clearer descriptions.

Fixes a bug in pkg/helm/environment/environment_test.go where environment variables were leaking from one test to the next.

Fixes #3456.

### CODE REVIEW

File: cmd/helm/helm.go [hunk 1]
```diff
@@ -75,6 +65,11 @@ Environment:
   $HELM_NO_PLUGINS    disable plugins. Set HELM_NO_PLUGINS=1 to disable plugins.
   $TILLER_NAMESPACE   set an alternative Tiller namespace (default "kube-system")
   $KUBECONFIG         set an alternative Kubernetes configuration file (default "~/.kube/config")
+  $HELM_TLS_CA_CERT   path to TLS CA certificate used to verify the Helm client and Tiller server certificates (default "$HELM_HOME/ca.pem")
```
Comment (stepin) [inline cmd/helm/helm.go]: maybe error here: comment says that `ca.pem` is default but there is no other (not deleted) `ca.pem` occurrences in this patch.

Comment (erwbgy) [reply cmd/helm/helm.go]: The default values are set in pkg/helm/helmpath/helmhome.go and the Init function in pkg/helm/environment/environment.go has been changed to use these - see "TLS defaults that depend on Home value".

**Comment by Nodraak** [comment]
hi !

What is the status here ? the --tls flag is a very bad user experience, and any solution (config file, env var, ...) would be a huge improvement

Do you need any help ?

**Comment by joaquinrdzibm** [comment]
any update on this PR?

**Comment by erwbgy** [comment]
@bacongobbler Any chance of a review?

**Comment by fetmar** [comment]
@erwbgy Looks like there's still a couple errors in the circle-ci build.

**Comment by bacongobbler** [comment]
yep! once CI's been fixed feel free to ping me and I can take another look.

**Comment by sporkmonger** [comment]
Isn't the CI fail just missing `make docs`?

**Comment by sporkmonger** [comment]
Sent @erwbgy a PR for his PR to fix the docs. :-P

**Review by michelleN — CHANGES_REQUESTED** [review]
Hey @erwbgy - would you mind fixing the merge conflicts? Let me know when you're ready and I'd be happy to review. 
@sporkmonger - Took a look just now and looks like the conflicts require more than just a `make docs` run.

Thank you!

**Comment by tavisma** [comment]
this would be wonderful to have! (as well as setting tiller namespace)

**Comment by bacongobbler** [comment]
Closing in favour of #4590
